### ISSUE TITLE
Delete old sync vfs code and cleanup

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2819]
|<img alt="cablehead's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| cablehead reported|
|-|-|
|Trac ID|trac#2819|
|Type|enhancement|
|Created|2007-09-06 10:13:10Z|

Once everything is ported to the new async interface, delete the old code and clean up module/variable names.

Depends on [#2816](https://github.com/twisted/twisted/issues/2816), [#2817](https://github.com/twisted/twisted/issues/2817), [#2818](https://github.com/twisted/twisted/issues/2818)

<details><summary>Searchable metadata</summary>

```
trac-id__2819 2819
type__enhancement enhancement
reporter__cablehead cablehead
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__vfs vfs
keywords__ 
time__1189073590000000 1189073590000000
changetime__1300038228000000 1300038228000000
version__None None
owner__cablehead cablehead

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="cablehead's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|cablehead commented|
|-|-|

Target layout:

```
twisted/vfs/__init__.py
twisted/vfs/ivfs.py
twisted/vfs/backends.py
twisted/vfs/adapters.py
twisted/vfs/plugin.py

twisted/vfs/test/test_backends.py
twisted/vfs/test/test_adapters.py
twisted/vfs/test/test_decorator.py
twisted/vfs/test/test_plugin.py
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set status to `closed`|
|-|-|

VFS has been removed ([#4934](https://github.com/twisted/twisted/issues/4934))
